one_liners2.cc

```diff
@@ -0,0 +1,56 @@
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <vector>
+#include <string>
+#include <boost/algorithm/string/classification.hpp>
+
+using namespace std;
+
+// 1. Xor to each char
+string xored_string(string s, char n) {
+    return accumulate(s.begin(), s.end(), string(""), [n](
+        const string& acc, const char& c){return acc + char(c^n);});
+}
+// reverse
+string my_reverse(string s) {
+    return accumulate(s.crbegin(), s.crend(), string(""), [](
+        const string& acc, const char& c){return acc + c;});
+}
+
+// check all capitals
+bool all_capitals(const vector<string>& names) {
+    return accumulate(names.begin(), names.end(), true, [](
+        bool acc, const string& name){
+            return acc & 
+                all_of(
+                    name.begin(), 
+                    name.end(), 
+                    boost::is_upper()
+                );});
+}           
+
+// flatten vector of vectors of int
+vector<int> flat_vec(const vector<vector<int>>& vectors) {
+    return accumulate(vectors.begin(), vectors.end(), vector<int>{},
+        [](vector<int> acc, const vector<int>& cur) {
+            move(cur.begin(), cur.end(), back_inserter(acc));
+            return acc;
```

Comment:
More than 1 line.

Reply:
It was specifically allowed to do 2 for this one.... :)

Reply:
oh, you're right!